Subject: [PATCH] Null pointer exception in FileRotator.java

Sometimes a null pointer exception is thrown when examining files
managed by the file rotator.

The logs from the test show that the Wifi state is changed a large
number of times. On every state change, a write operation is
initiated on the file system. This will eventually result in out
of memory and the call to mBasePath.list()method in FileRotator.java
will return null so the iteration will throw a NullPointerException.

diff --git a/core/java/com/android/internal/util/FileRotator.java b/core/java/com/android/internal/util/FileRotator.java
index 71550be..7320f37 100644
--- a/core/java/com/android/internal/util/FileRotator.java
+++ b/core/java/com/android/internal/util/FileRotator.java
@@ -306,7 +306,16 @@
         long oldestActiveStart = Long.MAX_VALUE;
 
         final FileInfo info = new FileInfo(mPrefix);
-        for (String name : mBasePath.list()) {
+
+        String[] baseFiles = mBasePath.list();
+        if (baseFiles == null) {
+            // no active file found above; create one starting now
+            info.startMillis = currentTimeMillis;
+            info.endMillis = Long.MAX_VALUE;
+            return info.build();
+        }
+
+        for (String name : baseFiles) {
             if (!info.parse(name)) continue;
 
             // pick the oldest active file which covers current time
